torchmetrics/detection/map.py

```diff
@@ -275,8 +298,10 @@ def __init__(
         self.add_state("detection_boxes", default=[], dist_reduce_fx=None)
         self.add_state("detection_scores", default=[], dist_reduce_fx=None)
         self.add_state("detection_labels", default=[], dist_reduce_fx=None)
+        self.add_state("detection_masks", default=[], dist_reduce_fx=None)
```

Comment:
if you allow only one kind of iou per class instance, there is no need to have all states here :)

Reply:
@justusschock  I followed the original API implementation of pycocotools, where you adopt only one iou_type